core/src/main/java/com/nisovin/magicspells/util/trackers/ParticleProjectileTracker.java

```diff
@@ -364,17 +362,19 @@ public void run() {
 			EulerAngle angle = EulerAngle.ZERO.setX(AccurateMath.toRadians(effectLoc.getPitch()));
 
 			if (armorStandSet != null) {
-				for (ArmorStand armorStand : armorStandSet) {
-					armorStand.teleportAsync(effectLoc);
-					armorStand.setHeadPose(angle);
-					armorStand.setLeftArmPose(angle);
-					armorStand.setRightArmPose(angle);
+				for (DelayableEntity<ArmorStand> armorStand : armorStandSet) {
+					armorStand.teleport(effectLoc);
+					armorStand.ifPresent(stand -> {
+						stand.setHeadPose(angle);
+						stand.setLeftArmPose(angle);
+						stand.setRightArmPose(angle);
+					});
```

Comment:
I wonder if it'd make sense to have another method such as `DelayableEntity#apply`, which uses `CompleteableFuture#thenApply` so that when it spawns these pose changes are still applied.

Reply:
I don't think it's too important here because in most cases the projectile has a low enough `tick-interval` that the incorrect spawn rotation wouldn't be noticed, however, I have applied the suggested changes by adding the `DelayableEntity#apply` method as well as exposing the future with `DelayableEntity#getFuture` so that `DelayableEntity` can be more useful in other cases.